app/forms/Invite_by_email_form.rb

```diff
@@ -0,0 +1,33 @@
+class InviteByEmailForm
+  include ActiveModel::Model
+  attr_accessor :email
+
+  DELIMITER = ','
+
+  validates :email,
+            presence: true,
+            format: {with: Devise.email_regexp, message: 'example:  john@gmail.com'}, unless: :email_is_list?
+  validate :emails_are_valid, if: :email_is_list?
+
+  private
+
+  def emails_are_valid
+    invalid_emails = emails.select do |email|
+      !InviteByEmailForm.new(email: email.strip).valid?
```

Comment:
I'd not use recursion, you can just compare string with regexp you have (`!email.strip.match(Devise.email_regexp)`)